src/components/Project/ProjectPage.tsx

```diff
@@ -1,15 +1,73 @@
 "use client";
 
-import { useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import ProjectTabs from "../ProjectTabs";
-import { ProjectTab } from "@/types/const";
+import { ProjectIDType, ProjectTab } from "@/types/const";
 import ProjectDetailsComponent from "../ProjectMetaDataTable/ProjectDetailsComponent";
 import ContractDetailsPage from "../ProjectContract/ContractDetailsPage";
 import ProjectDistribution from "./ProjectDistribution";
 import MockData from "./project.json";
+import { useAccount } from "wagmi";
+import getDecodedAttestationData from "@/lib/eas/getDecodedAttestationData";
+import { useProjectProvider } from "@/context/ProjectProvider";
+import { useParams } from "next/navigation";
+import { usePaymasterProvider } from "@/context/Paymasters";
+import { getProjects } from "@/lib/eas/getProjects";
 
 const ProjectPage = () => {
   const [tabContent, setTabContent] = useState<ProjectTab>("project");
+  const [mockData, setMockData] = useState(null);
+  const { address } = useAccount();
+  const { setName, setDescription } = useProjectProvider();
+  const { id } = useParams<ProjectIDType>();
+  const { error }: any = usePaymasterProvider();
+  let refUid = "";
+  let uid = id;
+  let accountAddress = address;
+
+  const getAddress = (projects: any) => {
+    if (projects?.data.length > 0) {
+      let refAttestation = projects.data[projects.data.length - 1];
+      refUid = refAttestation.result[5];
+      if (refUid === id) {
+        uid = refAttestation.result[0];
+        accountAddress = refAttestation.result[7];
+      }
+    }
+  };
```

Comment:
are you trying to use `refUid` as a state variable?

please do not update `let` variables...

in react, you should be using `useState` right?

Reply:
Im using them as variables didn't need to use the useState.

Reply:
why are you updating a variable outside the function instead of `state`?

https://stackoverflow.com/questions/67080000/react-when-to-use-global-variables-instead-of-states